### ISSUE TITLE
Wagtail Admin can't work without the `wagtail_serve` URL

### ISSUE THREAD

**Issue by m1kola** [wagtail/wagtail#4276]
### Issue Summary

It's possible to use Wagtail as a "headless CMS": you just create the admin UI for editing and server your content to front-end or a mobile application via Wagtail API.

In this case it doesn't make much sense to expose the `wagtail_serve` url in the application, but the problem is that Wagtail Admin UI fails if the `wagtail_serve` is undefined.

### Steps to Reproduce

1. Create a new Wagtail project or use existing
2. Remove `wagtail_serve` (`wagtail.wagtailcore.urls`) from your `urls.py`
3. Go to the page tree in the Admin UI via explorer menu

As a result you will get the `NoReverseMatch` exception on attempt to open the page tree view. I guess there are other places where Wagtail attempts to reverse `wagtail_serve`.

### Technical details

* Django version: 1.11
* Wagtail version: 1.13.1

### PS

I was going to classify this as a bug, but I looks like Wagtail documentation never mentioned (at least, I didn't manage to find this) that it's possible to use Wagtail without `wagtail_serve` or as a "headless CMS" at all. However we talk about this a lot, so would be great to make the admin UI to not depend on existence of front-end URLs.

I think, we should hide the UI elements like "live" links, if `wagtail_serve` is not defined in an application.

**Comment by thclark** [comment]
1.5 years on, the docs definitely do now mention using wagtail as a headless CMS... the torchbox founders are writing articles on how to do that, too.

For me the workaround was to include the wagtail urls *after* a catch-all URL which serves my frontend app bundle... so they never get touched, but the sitemap generates correctly... like this:
```

# Catch-all URL
# On any other url, including root, serve up the react-app and let react router take care of navigation within itself
urlpatterns += [url(r'^.*', ensure_csrf_cookie(ReactView.as_view(template_name=os.path.join(settings.REACT_DIR, 'build', 'index.html'))))]

# WORKAROUND:
# Wagtail doesn't run correctly in headless mode. It needs to have its urls installed to render the admin pages without 'no-reverse-match' errors.
# By placing them here, after the catch-all react app urls, we never actually use wagtail's serving mechanism, but wagtail's sitemap generator still works, assuming you use the same page slugs to route your frontend as you do in wagtail.
urlpatterns += [url(r'', include(wagtail_urls))]
```

It's unnerving to know that this must cause some broken reversals in the wagtail admin, though. Another way around it - if you don't care about wagtail generating a sitemap correctly (or are willing to strip this token out of any sitemap) - is to put them behind some long token like:
urlpatterns += [url(r'fliggertygibbet-never-ever-gonna-get-found-by-anyone', include(wagtail_urls))]

**Comment by gasman** [comment]
Fixed in #5827
